src/lib/battery/battery.h

```diff
@@ -37,54 +37,200 @@
+		param_t source;
+
+		// TODO: These parameters are depracated. They can be removed entirely once the
+		//  new version of Firmware has been around for long enough.
+		param_t v_empty_old;
+		param_t v_charged_old;
+		param_t n_cells_old;
+		param_t capacity_old;
+		param_t v_load_drop_old;
+		param_t r_internal_old;
+		param_t source_old;
+	} _param_handles;
+
+	struct {
+		float v_empty;
+		float v_charged;
+		int n_cells;
+		float capacity;
+		float v_load_drop;
+		float r_internal;
+		float low_thr;
+		float crit_thr;
+		float emergen_thr;
+		int source;
+
+		// TODO: These parameters are depracated. They can be removed entirely once the
+		//  new version of Firmware has been around for long enough.
+		float v_empty_old;
+		float v_charged_old;
+		int n_cells_old;
+		float capacity_old;
+		float v_load_drop_old;
+		float r_internal_old;
+		int source_old;
+	} _params;
+
+	battery_status_s _battery_status;
+
+	const int _index;
+
+	bool _first_parameter_update{false};
+	virtual void updateParams() override;
+
+	/**
+	 * This function helps with migrating to new parameters. It performs several tasks:
+	 *  - Update both the old and new parameter values using `param_get(...)`
+	 *  - Check if either parameter changed just now
+	 *    - If so, display a warning if the deprecated parameter was used
+	 *    - Copy the new value over to the other parameter
+	 *  - If this is the first time the parameters are fetched, check if they are equal
+	 *    - If not, display a warning and copy the value of the deprecated parameter over to the new one
+	 * @tparam T Type of the parameter (int or float)
+	 * @param old_param Handle to the old deprecated parameter (for example, param_find("BAT_N_CELLS")
+	 * @param new_param Handle to the new replacement parameter (for example, param_find("BAT1_N_CELLS")
+	 * @param old_val Pointer to the value of the old deprecated parameter
+	 * @param new_val Pointer to the value of the new replacement parameter
+	 * @param firstcall If true, then this function will not check to see if the values have changed
+	 * 					  (Since the old values are uninitialized)
+	 * @return True iff either of these parameters changed just now and the migration was done.
 	 */
-	void updateBatteryStatus(hrt_abstime timestamp, float voltage_v, float current_a,
-				 bool connected, bool selected_source, int priority,
-				 float throttle_normalized,
-				 bool armed, battery_status_s *status);
+	template<typename T>
+	bool migrateParam(param_t old_param, param_t new_param, T *old_val, T *new_val, bool firstcall)
+	{
+
+		T previous_old_val = *old_val;
+		T previous_new_val = *new_val;
+
+		param_get(old_param, old_val);
+		param_get(new_param, new_val);
+
+		if (!firstcall) {
+			if ((float) fabs((float) *old_val - (float) previous_old_val) > FLT_EPSILON
+			    && (float) fabs((float) *old_val - (float) *new_val) > FLT_EPSILON) {
+				// TODO fix this error message. I was getting hardfaults while using this message, but they stopped
```

Comment:
You can remove it completely, as a user would not see it.
The hardfault should either be a stack overflow or null-pointer dereference (`param_name`).